bin/runtime-common/src/messages.rs

```diff
@@ -344,71 +346,155 @@ pub mod target {
 	/// Verify proof of Bridged -> This chain messages.
 	pub fn verify_messages_proof<B: MessageBridge, ThisRuntime>(
 		proof: FromBridgedChainMessagesProof<B>,
+		max_messages: MessageNonce,
 	) -> Result<ProvedMessages<Message<BalanceOf<BridgedChain<B>>>>, &'static str>
```

Comment:
Since we have a proper error enum now, why not return it here and implement `Display` instead of `Into<&'static str>`?

Reply:
Implementing `Display` would bring things `from std::fmt` to runtime (like `Formatter`, `Result`, `write!`), right? I thought it isn't a good idea, even if possible.

Reply:
Oh yeah, good point, we don't want that.